docs/csharp/tutorials/upgrade-to-nullable-references.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Design with nullable reference types
+description: This advanced tutorial provides an introduction to nullable reference types. You'll learn to express your design intent on when reference values may be null, and have the compiler enforce when they cannot be null.
+ms.date: 12/03/2018
+ms.custom: mvc
+---
+# Tutorial: Migrate existing code with nullable reference types
+
+C# 8 introduces **nullable reference types**, which complement reference types the same way nullable value types complement value types. You declare a variable to be a **nullable reference type** by appending a `?` to the type. For example, `string?` represents a nullable `string`. You can use these new types to more clearly express your design intent: some variables *must always have a value*, others *may be missing a value*. Any existing variables of a reference type would be interpreted as a non-nullable reference type. 
+
+In this tutorial, you'll learn how to:
+
+> [!div class="checklist"]
+> * Enable null reference checks as you work with code.
+> * Diagnose and correct different warnings related to null values.
+> * Manage the interface between nullable enabled and nullable disabled contexts.
+> * Control nullable annotation contexts
+
+## Prerequisites
+
+You’ll need to set up your machine to run .NET Core, including the C# 8.0 beta compiler. The C# 8 beta compiler is available with [Visual Studio 2019 preview 1 and later](https://visualstudio.microsoft.com/vs/preview/), or [.NET Core 3.0 preview 1](https://dotnet.microsoft.com/download/dotnet-core/3.0).
+
+This tutorial assumes you're familiar with C# and .NET, including either Visual Studio or the .NET Core CLI.
+
+The sample application that you'll migrate is an RSS feed reader web app. It reads from a single RSS feed and displays summaries for the most recent articles. You can click on any of the articles to visit the site. The application is relatively new, but was written before nullable reference types were available. The design decisions for the application represented sound principles, but don't take advantage of this important language feature.
+
+The sample application includes a unit test library that validates the major functionality of the app. That project will make it easier to upgrade safely, if you change any of the implementation based on the warnings generated. You can download the starter code from our [samples GitHub repo](https://github.com/dotnet/samples/tree/master/csharp/tutorials/nullable-reference-migration/start).
+
+Your goal migrating a project should be to leverage the new language features so that you clearly express your intent on the nullability of variables, and do so in such a way that the  compiler does not generate warnings when you have the nullable annotation context and nullable warning context set to `enabled`.
```

Comment:
```suggestion
Your goal migrating a project should be to leverage the new language features so that you clearly express your intent on the nullability of variables, and do so in such a way that the compiler doesn't generate warnings when you have the nullable annotation context and nullable warning context set to `enabled`.
```